include/sys/nvpair.h

```diff
@@ -232,7 +232,8 @@ _SYS_NVPAIR_H int nvlist_lookup_int64(const nvlist_t *, const char *,
     int64_t *);
 _SYS_NVPAIR_H int nvlist_lookup_uint64(const nvlist_t *, const char *,
     uint64_t *);
-_SYS_NVPAIR_H int nvlist_lookup_string(nvlist_t *, const char *, char **);
+_SYS_NVPAIR_H int nvlist_lookup_string(const nvlist_t *, const char *,
+    const char **);
```

Comment:
These API changes would mean we're diverging from the long standing [documented interfaces](https://docs.oracle.com/cd/E36784_01/html/E36878/nvlist-lookup-string-3nvpair.html).   Historically we've tried to avoid that kind of thing.  However, now that we're converging on a single code base it's a bit less problematic.  Given the upside of this change I'm okay with it.

Reply:
I generally shy away from divergence, but I make an exception for const correctness. Despite libabigail’s claims to the contrary, the binary ABI is the same and programs will still compile without -Werror, although they could have warnings. That said, the only places where we would have those problems are in the f versions of these functions that is not part of the OpenSolaris DDI and the functions who had handles constified. The benefits of constification outweighs the downsides and an earlier commit already constified some of the libnvpair functions.

Reply:
One additional thing to note for future reference is that compilers are always adding new warnings about correctness that similarly break programs built with -Werror. The effect of constification is not dissimilar to them.